src/python/pants/backend/python/rules/python_create_binary.py

```diff
@@ -0,0 +1,75 @@
+# Copyright 2019 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from pants.backend.python.rules.inject_init import InjectedInitDigest
+from pants.backend.python.rules.pex import CreatePex, Pex, PexInterpreterContraints, PexRequirements
+from pants.backend.python.subsystems.python_setup import PythonSetup
+from pants.backend.python.targets.python_binary import PythonBinary
+from pants.engine.fs import Digest, DirectoriesToMerge
+from pants.engine.isolated_process import ExecuteProcessRequest, FallibleExecuteProcessResult
+from pants.engine.legacy.graph import BuildFileAddresses, HydratedTarget, TransitiveHydratedTargets
+from pants.engine.legacy.structs import PythonBinaryAdaptor
+from pants.engine.rules import UnionRule, rule
+from pants.engine.selectors import Get
+from pants.rules.core.binary import BinaryTarget, CreatedBinary
+from pants.rules.core.strip_source_root import SourceRootStrippedSources
+
+
+@rule
+def create_python_binary(python_binary_adaptor: PythonBinaryAdaptor,
+  python_setup: PythonSetup) -> CreatedBinary:
+  transitive_hydrated_targets = yield Get(
+    TransitiveHydratedTargets, BuildFileAddresses((python_binary_adaptor.address,))
+  )
+  all_targets = transitive_hydrated_targets.closure
+  all_target_adaptors = [t.adaptor for t in all_targets]
+
+
+  interpreter_constraints = PexInterpreterContraints.create_from_adaptors(
+    adaptors=tuple(all_targets),
+    python_setup=python_setup
+  )
+
+  source_root_stripped_sources = yield [
+    Get(SourceRootStrippedSources, HydratedTarget, target_adaptor)
+    for target_adaptor in all_targets
+  ]
+
+  #TODO This way of calculating the entry point works but is a bit hackish.
```

Comment:
Would you mind expanding on this before landing? or opening a ticket?

Reply:
Opened a ticket for this.